Fails to compile on new macs?

A user reported this on one of the new m1 macs:

```
error[E0277]: the size for values of type `str` cannot be known at compilation time
--> /Users/s/.cargo/registry/src/github.com-1ecc6299db9ec823/bitcoind-0.17.0/build.rs:50:9
|
50 | / println!(
51 | | "filename:{} version:{} hash:{}",
52 | | download_filename, VERSION, expected_hash
53 | | );
| |__________^ doesn't have a size known at compile-time
|
= help: the trait `Sized` is not implemented for `str`
= note: this error originates in the macro `$crate::format_args_nl` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the size for values of type `str` cannot be known at compilation time
--> /Users/s/.cargo/registry/src/github.com-1ecc6299db9ec823/bitcoind-0.17.0/build.rs:55:19
|
55 | let url = format!(
| ___________________^
56 | | "https://bitcoincore.org/bin/bitcoin-core-{}/{}",
57 | | VERSION, download_filename
58 | | );
| |_________^ doesn't have a size known at compile-time
|
= help: the trait `Sized` is not implemented for `str`
= note: this error originates in the macro `$crate::__export::format_args` (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 3 previous errors

Some errors have detailed explanations: E0277, E0425.
For more information about an error, try `rustc --explain E0277`.
error: could not compile `bitcoind`

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

Can anyone confirm or deny?

--- a/build.rs
+++ b/build.rs
@@ -8,7 +8,10 @@ use tar::Archive;
 
 include!("src/versions.rs");
 
-#[cfg(all(target_os = "macos", target_arch = "x86_64"))]
+#[cfg(all(
+    target_os = "macos",
+    any(target_arch = "x86_64", target_arch = "aarch64")
+))]
 fn download_filename() -> String {
     format!("bitcoin-{}-osx64.tar.gz", &VERSION)
 }

